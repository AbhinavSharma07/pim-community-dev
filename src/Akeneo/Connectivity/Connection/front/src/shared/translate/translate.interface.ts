export type Translate = (id: string, placeholders?: any, count?: number) => string;
