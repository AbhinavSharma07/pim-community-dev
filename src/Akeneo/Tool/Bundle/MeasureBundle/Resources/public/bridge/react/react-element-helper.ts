import ReactDOM from 'react-dom';

let container: Element | null = null;

const mountReactElementRef = (component: JSX.Element) => {
  if (null === container) {
    container = document.createElement('div');
    // (container as any).style.height = '100%';
    // (container as any).style.display = 'flex';
    // (container as any).style.flexDirection = 'column';
    ReactDOM.render(component, container);
  }

  return container;
};

const unmountReactElementRef = () => {
  if (null !== container) {
    ReactDOM.unmountComponentAtNode(container);
    container = null;
  }
};

export {mountReactElementRef, unmountReactElementRef};
