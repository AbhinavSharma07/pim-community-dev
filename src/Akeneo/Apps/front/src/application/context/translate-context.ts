import * as React from 'react';
import {translate} from '../service/shared/translate.interface';

export const TranslateContext = React.createContext<translate>(id => id);
