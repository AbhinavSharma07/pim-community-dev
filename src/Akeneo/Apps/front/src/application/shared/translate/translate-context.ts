import * as React from 'react';
import {Translate} from './translate.interface';

export const TranslateContext = React.createContext<Translate>(id => id);
