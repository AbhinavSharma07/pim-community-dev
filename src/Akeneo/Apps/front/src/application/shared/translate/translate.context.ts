import * as React from 'react';
import {translate} from './translate.interface';

export const TranslateContext = React.createContext<translate>(id => id);
