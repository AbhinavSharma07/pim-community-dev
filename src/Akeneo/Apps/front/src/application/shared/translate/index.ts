import {Translate} from './Translate';
import {TranslateContext} from './translate-context';
import {Translate as TranslateInterface} from './translate.interface';

export {TranslateContext, Translate, TranslateInterface};
