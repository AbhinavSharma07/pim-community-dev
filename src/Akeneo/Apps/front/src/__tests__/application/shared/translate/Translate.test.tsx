import * as React from 'react';
import {create} from 'react-test-renderer';
import {TranslateContext} from '../../../../application/shared/translate/translate-context';
import {Translate} from '../../../../application/shared/translate/Translate';

describe('Translate component', () => {
    test('Matches the snapshot', () => {
        const translate = () => 'Bonjour le monde';
        const component = create(
            <TranslateContext.Provider value={translate}>
                <Translate id='hello.world' />
            </TranslateContext.Provider>
        );

        expect(component.toJSON()).toMatchSnapshot();
    });
});
