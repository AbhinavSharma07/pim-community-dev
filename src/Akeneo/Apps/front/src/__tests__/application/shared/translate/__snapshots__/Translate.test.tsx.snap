// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Translate component Matches the snapshot 1`] = `"Bonjour le monde"`;
